### PR TITLE
Editing ButtonSet documentation

### PR DESCRIPTION
**Pull request by heatherlarsen**
### :pushpin: Summary

If merged, this PR will update the documentation for the ButtonSet component.

**Preview:** https://hds-website-git-heatherlarsen-button-set-docs-hashicorp.vercel.app/components/button-set/

### :link: External links

Jira ticket: [HDS-1205](https://hashicorp.atlassian.net/browse/HDS-1205)

***

### 👀 Reviewer's checklist:

- [ ] +1 Percy if applicable
- [ ] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

File: website/docs/components/button-set/index.md [hunk 1]
```diff
@@ -2,22 +2,18 @@
 title: ButtonSet
 description: Provides consistent layout and spacing for a set of buttons.
 caption: A set of buttons.
-status: released
+status: code-only
```
[the remark this answers is not in the document]
Comment (heatherlarsen) [reply website/docs/components/button-set/index.md:5]: A ButtonSet component does not exist in Figma; therefore, it's only available in the code. Whereas "released" is available in both Figma and Code.

[the remark this answers is not in the document]
Comment (KristinLBradley) [reply website/docs/components/button-set/index.md:5]: I see, thanks for explaining

**Review by KristinLBradley — APPROVED** [review]

**Review by jorytindall — APPROVED** [review]
I think this is fine, but I also think that the guidelines are useful and important for designers. We have some basic content right now, additional content from my research project might pair well with this.

**Comment by heatherlarsen** [comment]
@jorytindall I don't disagree, and it would be great if you could add guidance to this component based on your research. We don't currently have anything meaningful documented for a button set within the Figma file.

**Pull request merged**